packages/rancher-sachet/charts/templates/config-map.yaml

```diff
@@ -0,0 +1,127 @@
+{{- if and (.Release.IsInstall) (not (lookup "v1" "ConfigMap" .Values.namespace (include "sachet.fullname" .))) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: {{ .Values.namespace }}
+  {{- include "sachet.preNameLabelAnno" . | nindent 2 }}
+data:
+  config.yaml: |-
+    {{- with .Values.sachet.providers }}
+    providers:
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
+    {{- with .Values.sachet.receivers }}
+    receivers:
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
+    {{- $template := $.Files.Get "files/template.tmpl" }}
+    {{- if $template }}
+    template.tmpl: |-
+      {{- $template | nindent 6 }}
+    {{- end }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: {{ .Values.namespace }}
+  {{- include "sachet.preNameLabel" . | nindent 2 }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": hook-succeeded,hook-failed,before-hook-creation
+    "helm.sh/hook-weight": "5"
+spec:
+  template:
+    metadata:
+      {{- include "sachet.preNameLabel" . | nindent 6 }}
+    spec:
+      serviceAccountName: {{ include "sachet.fullname" . }}-pre-install
+      {{- if .Values.preInstallHook.securityContext }}
+      securityContext:
+      {{- toYaml .Values.preInstallHook.securityContext | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: copy-pre-install-configmap
+          image: {{ include "system_default_registry" . }}{{ .Values.preInstallHook.image.repository }}:{{ .Values.preInstallHook.image.tag }}
+          imagePullPolicy: {{ .Values.preInstallHook.image.pullPolicy }}
+          command:
+            - /bin/sh
+            - -c
+            - >
+              if kubectl get configmap -n {{ .Values.namespace }} {{ include "sachet.fullname" . }} > /dev/null 2>&1; then
+                echo "configMap already exists"
+                exit 1
+              fi;
+              kubectl patch configmap -n {{ .Values.namespace }} --dry-run -o yaml {{ include "sachet.preName" . }}
+              -p '{{ printf "{\"metadata\":{\"name\": \"%s\"}}" (include "sachet.fullname" .) }}'
+              | kubectl apply -f -;
+              kubectl annotate configmap -n {{ .Values.namespace }} {{ include "sachet.preName" . }}
+              helm.sh/hook- helm.sh/hook-delete-policy- helm.sh/hook-weight-;
+      restartPolicy: OnFailure
+      nodeSelector: {{ include "linux-node-selector" . | nindent 8 }}
+      tolerations: {{ include "linux-node-tolerations" . | nindent 8 }}
```

Comment:
Just trying to understand here: what is the purpose of this pre-install hook?

Reply:
It is to make the configmap independent from the app, ie. uninstalling/updating the app will NOT affect the configmap, so we can preserve the user's configuration.

Reply:
Hmmm, alright that makes sense. Pre/post install hooks are usually "last resort" options for me, but this looks like a great use case! I'll resolve this.